README.md

```diff
@@ -9,6 +9,14 @@ The Node.js project uses an [open governance model](./GOVERNANCE.md). The
 
 **This project has a [Code of Conduct][].**
 
+In addition to the code of conduct requirements, contributors to the
+Node.js project are expected to act in a collaborative manner to move
+the project forward. We encourage the constructive exchange of contrary
+opinions and compromise. Contributors who repeatedly act in ways that
+discourage, exhaust, or otherwise negatively affect other contributors
+will be limited in the roles they are allowed to hold or blocked
+from the project completely.
```

Comment:
This seems a bit arbitrary. Who is going to decide this and how?

Reply:
Based on the discussions it will be the responsibilty of the TSC to make such decisions.

Reply:
By a vote? Simple majority? Something else?

Will such a vote be public or private?

Are warnings utilized here or no? What recourse do contributors have (if any)?

Reply:
I think the text is deliberately vague because situations will be handled on a case-by-case basis. Generally anything concerning personnel is conducted in private.

Reply:
> I think the text is deliberately vague because situations will be handled on a case-by-case basis.

That's not very reassuring that everyone's case will be treated the same.

Reply:
The TSC isn't a court. Its job is to manage the project effectively.

Reply:
> That's not very reassuring that everyone's case will be treated equally.

Long-winded hot take follows. Sorry/not sorry.

It's a good idea not to conflate related but distinct issues:

* Are people being treated fairly under a consistent process?
* Do people feel like they are being treated fairly under a consistent process?
* Do observers feel like people are being treated fairly under a consistent process? 

The other concerns to prioritize are:

* What's best for Node.js contributors and users?
* What process takes the least toll/does the least damage to the TSC and other maintainers?

Of those five concerns, "Do people feel like they are being treated fairly?" is (for me) the least important. It is important, but vastly less important than those last two or the first one. It is also very hard to achieve. People who are subject to censure rarely feel that they have been treated fairly no matter how elaborate and lengthy the process, no matter how much bending-over-backwards and second-chances are supplied, etc. 

It's tempting to think that observer perceptions are substantially more important, but they're not, mostly because they tend to fall along the lines of how the individuals feel. The friends/allies of recipients of censure will invariably feel that they are being treated poorly/unfairly, and their critics/accusers will never be satisfied with a process that is slow or that vindicates the accused. There's just no winning, so we're opting out of playing. We can come up with an elaborate process with appeals and we can become ineffective and slow and create a lot of work for ourselves, all for little or no ROI. Or we can admit the obvious: The TSC is always free to make decisions about this kind of stuff in any way they see fit.

If you have concrete suggestions for a process that would improve on the perceived shortcomings here, whether that's lack of transparency or assurances of consistency/fairness or anything else, I'm open to them. But honestly, again, no matter what is written here, the bottom line is that the TSC is always free to make decisions about this kind of stuff in any way they see fit. The only restrictions are in the charter, the law, and perhaps social opprobrium. This text just makes that implicit truth explicit.